File: SDKs/googleapis.com/youtubereporting/v1/java/lib/src/main/java/openapisdk/models/shared/GdataCompositeMedia.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * GdataCompositeMedia
 * gdata
**/
public class GdataCompositeMedia {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("blobRef")
    public String blobRef;
    public GdataCompositeMedia withBlobRef(String blobRef) {
        this.blobRef = blobRef;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("blobstore2Info")
    public GdataBlobstore2Info blobstore2Info;
    public GdataCompositeMedia withBlobstore2Info(GdataBlobstore2Info blobstore2Info) {
        this.blobstore2Info = blobstore2Info;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("cosmoBinaryReference")
    public String cosmoBinaryReference;
    public GdataCompositeMedia withCosmoBinaryReference(String cosmoBinaryReference) {
        this.cosmoBinaryReference = cosmoBinaryReference;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("crc32cHash")
    public Long crc32cHash;
    public GdataCompositeMedia withCrc32cHash(Long crc32cHash) {
        this.crc32cHash = crc32cHash;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("inline")
    public String inline;
    public GdataCompositeMedia withInline(String inline) {
        this.inline = inline;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("length")
    public String length;
    public GdataCompositeMedia withLength(String length) {
        this.length = length;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("md5Hash")
    public String md5Hash;
    public GdataCompositeMedia withMd5Hash(String md5Hash) {
        this.md5Hash = md5Hash;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("objectId")
    public GdataObjectId objectId;
    public GdataCompositeMedia withObjectId(GdataObjectId objectId) {
        this.objectId = objectId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("path")
    public String path;
    public GdataCompositeMedia withPath(String path) {
        this.path = path;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("referenceType")
    public GdataCompositeMediaReferenceTypeEnum referenceType;
    public GdataCompositeMedia withReferenceType(GdataCompositeMediaReferenceTypeEnum referenceType) {
        this.referenceType = referenceType;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("sha1Hash")
    public String sha1Hash;
    public GdataCompositeMedia withSha1Hash(String sha1Hash) {
        this.sha1Hash = sha1Hash;
        return this;
    }
}

File: SDKs/googleapis.com/youtubereporting/v1/python/src/sdk/models/shared/gdatacompositemedia.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import gdatablobstore2info as shared_gdatablobstore2info
from ..shared import gdataobjectid as shared_gdataobjectid

class GdataCompositeMediaReferenceTypeEnum(str, Enum):
    PATH = "PATH"
    BLOB_REF = "BLOB_REF"
    INLINE = "INLINE"
    BIGSTORE_REF = "BIGSTORE_REF"
    COSMO_BINARY_REFERENCE = "COSMO_BINARY_REFERENCE"


@dataclass_json
@dataclasses.dataclass
class GdataCompositeMedia:
    r"""GdataCompositeMedia
    gdata
    """
    
    blob_ref: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('blobRef') }})
    blobstore2_info: Optional[shared_gdatablobstore2info.GdataBlobstore2Info] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('blobstore2Info') }})
    cosmo_binary_reference: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cosmoBinaryReference') }})
    crc32c_hash: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('crc32cHash') }})
    inline: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('inline') }})
    length: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('length') }})
    md5_hash: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('md5Hash') }})
    object_id: Optional[shared_gdataobjectid.GdataObjectID] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('objectId') }})
    path: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('path') }})
    reference_type: Optional[GdataCompositeMediaReferenceTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('referenceType') }})
    sha1_hash: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sha1Hash') }})
    
